File: customerapp/src/main/java/com/tokopedia/tkpd/home/wishlist/adapter/viewmodel/WishlistRecomendationViewModel.java
package com.tokopedia.tkpd.home.wishlist.adapter.viewmodel;

import com.tokopedia.abstraction.base.view.adapter.Visitable;
import com.tokopedia.recommendation_widget_common.presentation.model.RecommendationItem;
import com.tokopedia.tkpd.home.wishlist.adapter.factory.WishlistTypeFactory;

/**
 * Author errysuprayogi on 25,July,2019
 */
public class WishlistRecomendationViewModel implements Visitable<WishlistTypeFactory> {

    private RecommendationItem recommendationItem;

    public WishlistRecomendationViewModel(RecommendationItem recommendationItem) {
        this.recommendationItem = recommendationItem;
    }

    public RecommendationItem getRecommendationItem() {
        return recommendationItem;
    }

    @Override
    public int type(WishlistTypeFactory typeFactory) {
        return typeFactory.type(this);
    }
}

File: customerapp/src/main/java/com/tokopedia/tkpd/home/adapter/viewmodel/WishlistRecomendationViewModel.kt
package com.tokopedia.tkpd.home.adapter.viewmodel

import com.tokopedia.abstraction.base.view.adapter.Visitable
import com.tokopedia.recommendation_widget_common.presentation.model.RecommendationItem
import com.tokopedia.tkpd.home.adapter.factory.WishlistTypeFactory

/**
 * Author errysuprayogi on 03,July,2019
 */
class WishlistRecomendationViewModel(val recommendationItem: RecommendationItem) : Visitable<WishlistTypeFactory> {

    override fun type(typeFactory: WishlistTypeFactory): Int {
        return typeFactory.type(this)
    }
}
